src/models/message.ts

```diff
@@ -75,11 +75,11 @@ export default class Message extends RestfulModel implements MessageProperties {
     return Object.values(participants);
   }
 
-  fileIds() {
+  fileIds(): (string | undefined)[] {
     return this.files ? this.files.map(file => file.id) : [];
   }
```

Comment:
@mrashed-dev should this really be an array of `string | undefined` or can we filter out `undefined` ids? Not sure if there is a requirement to get the same # of array elements even if we don't have valid IDs.
```suggestion
  fileIds(): (string | undefined)[] {
    return this.files ? this.files.filter(file => file.id).map(file => file.id) : [];
  }
```

Reply:
@AaronDDM I think that it will always return `string | undefined` because `this.files` is not guaranteed to be populated, which I think means it yields an undefined array type.